From 7e02f1cf3dea5c5423563b60ebe25daaeb8b0118 Mon Sep 17 00:00:00 2001
From: Julien Jerphanion <git@jjerphan.xyz>
Date: Wed, 7 Jan 2026 12:38:13 +0100
Subject: [PATCH] build: Use `mamba` namespace for targets

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
---
 libmamba-spdlog/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libmamba-spdlog/CMakeLists.txt b/libmamba-spdlog/CMakeLists.txt
index 249b36c3..a6cf73a7 100644
--- a/libmamba-spdlog/CMakeLists.txt
+++ b/libmamba-spdlog/CMakeLists.txt
@@ -60,12 +60,12 @@ set(libmamba_spdlog_targets "")
 
 if(BUILD_SHARED)
     message(STATUS "Adding target libmamba-spdlog with shared libmamba")
-    libmamba_spdlog_create_target(libmamba-dyn-spdlog SHARED libmamba-dyn)
+    libmamba_spdlog_create_target(libmamba-dyn-spdlog SHARED mamba::libmamba-dyn)
 endif()
 
 if(BUILD_STATIC)
     message(STATUS "Adding target libmamba-spdlog with static libmamba")
-    libmamba_spdlog_create_target(libmamba-static-spdlog STATIC libmamba-static)
+    libmamba_spdlog_create_target(libmamba-static-spdlog STATIC mamba::libmamba-static)
 endif()
 
 if(BUILD_SHARED)
-- 
2.52.0

