From 3044ba2ac2fb835e8621208ed2d1482260a2e03b Mon Sep 17 00:00:00 2001
From: Julien Jerphanion <git@jjerphan.xyz>
Date: Tue, 7 Apr 2026 16:41:39 +0200
Subject: [PATCH] fix: Adapt bindings for `UnresolvedChannel::location`

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
---
 libmambapy/bindings/specs.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/libmambapy/bindings/specs.cpp b/libmambapy/bindings/specs.cpp
index cb6adeb6..8b951392 100644
--- a/libmambapy/bindings/specs.cpp
+++ b/libmambapy/bindings/specs.cpp
@@ -345,7 +345,10 @@ namespace mambapy
             .def("__copy__", &copy<UnresolvedChannel>)
             .def("__deepcopy__", &deepcopy<UnresolvedChannel>, py::arg("memo"))
             .def_property_readonly("type", &UnresolvedChannel::type)
-            .def_property_readonly("location", &UnresolvedChannel::location)
+            .def_property_readonly(
+                "location",
+                [](const UnresolvedChannel& self) -> std::string { return self.location(); }
+            )
             .def_property_readonly("platform_filters", &UnresolvedChannel::platform_filters);
 
         py::class_<BasicHTTPAuthentication>(m, "BasicHTTPAuthentication")
-- 
2.53.0

