From b49f8b51d4548a5c3d1d7f28eb1f1e28ecf9958a Mon Sep 17 00:00:00 2001
From: Johan Mabille <johan.mabille@gmail.com>
Date: Wed, 7 Jan 2026 11:28:08 +0100
Subject: [PATCH] Added missing find_package(libmamba-spdlog) to micromamba
 CMakeLists

---
 micromamba/CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/micromamba/CMakeLists.txt b/micromamba/CMakeLists.txt
index 80ec642ff8..af213d4662 100644
--- a/micromamba/CMakeLists.txt
+++ b/micromamba/CMakeLists.txt
@@ -83,6 +83,9 @@ macro(mambaexe_create_target target_name linkage output_name)
         if(NOT (TARGET mamba::libmamba-static))
             find_package(libmamba REQUIRED)
         endif()
+        if(NOT (TARGET mamba::libmamba-static-spdlog))
+            find_package(libmamba-spdlog REQUIRED)
+        endif()
         target_link_libraries(
             ${target_name} PRIVATE mamba::libmamba-static mamba::libmamba-static-spdlog
         )
@@ -95,6 +98,9 @@ macro(mambaexe_create_target target_name linkage output_name)
         if(NOT (TARGET mamba::libmamba-dyn))
             find_package(libmamba REQUIRED)
         endif()
+        if(NOT (TARGET mamba::libmamba-dyn-spdlog))
+            find_package(libmamba-spdlog REQUIRED)
+        endif()
         target_link_libraries(${target_name} PRIVATE mamba::libmamba-dyn mamba::libmamba-dyn-spdlog)
     endif()
 
