diff --git a/libmamba/libmambaConfig.cmake.in b/libmamba/libmambaConfig.cmake.in
index d5260585..7c3a2b06 100644
--- a/libmamba/libmambaConfig.cmake.in
+++ b/libmamba/libmambaConfig.cmake.in
@@ -27,6 +27,8 @@ find_dependency(nlohmann_json)
 find_dependency(spdlog)
 find_dependency(Threads)
 find_dependency(tl-expected)
+find_dependency(zstd)
+find_dependency(BZip2)
 
 if(NOT (TARGET libmamba OR TARGET libmamba-static OR TARGET libmamba-full-static))
     include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake")
diff --git a/libmambapy/CMakeLists.txt b/libmambapy/CMakeLists.txt
index d743dd6f..b7f7c9d8 100644
--- a/libmambapy/CMakeLists.txt
+++ b/libmambapy/CMakeLists.txt
@@ -11,9 +11,6 @@ cmake_policy(SET CMP0057 NEW)
 
 project(libmambapy)
 
-find_package(fmt REQUIRED)
-find_package(spdlog REQUIRED)
-
 if(NOT TARGET libmamba)
     find_package(libmamba REQUIRED)
 endif ()
