diff --git a/include/solver.cpp b/include/solver.cpp
index d338f72..2651b97 100644
--- a/include/solver.cpp
+++ b/include/solver.cpp
@@ -5,7 +5,8 @@
 
 #include "solver.hpp"
 
-#ifndef _MSC_VER
+// we can't use C++17 on OS X right now
+#if !defined(_MSC_VER) || !defined(__APPLE__)
     #include <string_view>
     using std::string_view;
 #else
diff --git a/setup.py b/setup.py
index 952247b..a910cb7 100644
--- a/setup.py
+++ b/setup.py
@@ -92,10 +92,14 @@ class BuildExt(build_ext):
         ct = self.compiler.compiler_type
 
         if sys.platform == 'darwin':
-            self.c_opts['unix'] += ['-stdlib=libc++', '-mmacosx-version-min=10.7']
-            if not has_flag(self.compiler, '-std=c++17'):
-                self.c_opts['unix'].remove('-std=c++17')
-                self.c_opts['unix'].append('-std=c++1z')
+            self.c_opts['unix'] += ['-stdlib=libc++']
+            # if not has_flag(self.compiler, '-std=c++17'):
+            #    self.c_opts['unix'].remove('-std=c++17')
+            #    self.c_opts['unix'].append('-std=c++1z')
+
+            # we currently have to use C++14 to work around some pybind11 issues on old OS X versions
+            self.c_opts['unix'].remove('-std=c++17')
+            self.c_opts['unix'].append('-std=c++14')
 
         opts = self.c_opts.get(ct, [])
         if ct == 'unix':
