diff --git a/include/package_paths.hpp b/include/package_paths.hpp
index 9b0cde8..4d39ad4 100644
--- a/include/package_paths.hpp
+++ b/include/package_paths.hpp
@@ -6,10 +6,10 @@
 #include <map>
 #include <set>
 
-const std::string PREFIX_PLACEHOLDER("/opt/anaconda1anaconda2"
+const std::string PREFIX_PLACEHOLDER_1("/opt/anaconda1anaconda2");
                                      // this is intentionally split into parts, such that running
                                      // this program on itself will leave it unchanged
-                                     "anaconda3");
+const std::string PREFIX_PLACEHOLDER_2("anaconda3");
 
 namespace mamba
 {
diff --git a/src/package_paths.cpp b/src/package_paths.cpp
index 237f352..2b8c9dc 100644
--- a/src/package_paths.cpp
+++ b/src/package_paths.cpp
@@ -31,7 +31,7 @@ namespace mamba
             auto s = split(l, " ");
             if (s.size() == 1)
             {
-                res[s[0]] = PrefixFileParse{PREFIX_PLACEHOLDER, "text", s[0]};
+                res[s[0]] = PrefixFileParse{concat(PREFIX_PLACEHOLDER_1, PREFIX_PLACEHOLDER_2), "text", s[0]};
             }
             else if (s.size() == 3)
             {
