diff --git a/include/fetch.hpp b/include/fetch.hpp
index fa975a6..cf63029 100644
--- a/include/fetch.hpp
+++ b/include/fetch.hpp
@@ -160,15 +160,15 @@ namespace mamba
                 value = header.substr(colon_idx, header.size() - colon_idx - 2);
                 if (key == "ETag")
                 {
-                    s->etag = value;
+                    s->etag = std::string(value);
                 }
                 else if (key == "Cache-Control")
                 {
-                    s->cache_control = value;
+                    s->cache_control = std::string(value);
                 }
                 else if (key == "Last-Modified")
                 {
-                    s->mod = value;
+                    s->mod = std::string(value);
                 }
             }
             return nitems * size;
diff --git a/include/output.hpp b/include/output.hpp
index 4706af0..153529c 100644
--- a/include/output.hpp
+++ b/include/output.hpp
@@ -113,13 +113,18 @@ namespace mamba
             {
             }
 
+            SpecialStream(SpecialStream&& rhs)
+                : std::stringstream(std::move(rhs))
+            {
+            }
+
             ~SpecialStream()
             {
                 Output::instance().print(str());
             }
         };
 
-        static SpecialStream print()
+        static auto print()
         {
             return SpecialStream();
         }
diff --git a/include/query.hpp b/include/query.hpp
index 5b6a49e..1deb701 100644
--- a/include/query.hpp
+++ b/include/query.hpp
@@ -1,7 +1,12 @@
 #pragma once
 
 #include <sstream>
+#ifdef _WIN32
+#include "thirdparty/string_view.hpp"
+namespace std { using nonstd::string_view; }
+#else
 #include <string_view>
+#endif
 
 #include "pool.hpp"
 
diff --git a/include/util.hpp b/include/util.hpp
index dab2c65..3cf1505 100644
--- a/include/util.hpp
+++ b/include/util.hpp
@@ -1,11 +1,17 @@
 #pragma once
 
 #include <stdexcept>
-#include <string_view>
 #include <iostream>
 #include <iomanip>
 #include <cassert>
 
+#ifdef _WIN32
+#include "thirdparty/string_view.hpp"
+namespace std { using nonstd::string_view; }
+#else
+#include <string_view>
+#endif
+
 #include "thirdparty/filesystem.hpp"
 namespace fs = ghc::filesystem;
 
diff --git a/setup.py b/setup.py
index 33cd2a2..7f4a826 100644
--- a/setup.py
+++ b/setup.py
@@ -91,7 +91,7 @@ def has_flag(compiler, flagname):
 class BuildExt(build_ext):
     """A custom build extension for adding compiler-specific options."""
     c_opts = {
-        'msvc': ['/EHsc', '/std:c++17', '/Ox'],
+        'msvc': ['/EHsc', '/std:c++14', '/Ox'],
         'unix': ['-std=c++17', '-O3'],
     }
 
