diff --git a/libmamba/libmambaConfig.cmake.in b/libmamba/libmambaConfig.cmake.in
index 743fd1e9..e927a133 100644
--- a/libmamba/libmambaConfig.cmake.in
+++ b/libmamba/libmambaConfig.cmake.in
@@ -29,6 +29,7 @@ find_dependency(Threads)
 find_dependency(tl-expected)
 find_dependency(zstd)
 find_dependency(BZip2)
+find_dependency(yaml-cpp)
 
 if(NOT (TARGET libmamba OR TARGET libmamba-static))
     include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake")
